### PR TITLE
Added Just in time installation in Message extension Action JavaScript/Node JS

### PR DESCRIPTION
**Pull request by Trinetra-MSFT**
Added JIT installation in ME Action Node Js sample code.
PR C#: #3012 

Existing Node PR: #3047

## Proposed Changes
  -  Added Bot command Fetch roaster for Just in time installation
  - Updated Manifest

## Testing
Invoke Fetch Roaster command from ME
Get Just in time installation card
image

### CODE REVIEW

File: samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js [hunk 1]
```diff
@@ -1,45 +1,134 @@
...
     // The user has chosen to share a message by choosing the 'Share Message' context menu command.
     let userName = 'unknown';
-    if (action.messagePayload.from &&
-            action.messagePayload.from.user &&
-            action.messagePayload.from.user.displayName) {
+    if (
+        action.messagePayload.from &&
+        action.messagePayload.from.user &&
+        action.messagePayload.from.user.displayName
+    ) {
```
Comment (mdrichardson) [inline samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js:131]: Should be able to simplify to:

```suggestion
    if (action.messagePayload?.from?.user?.displayName) {
```

File: samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js [hunk 2]
```diff
@@ -48,28 +137,37 @@ function shareMessageCommand(context, action) {
...
         action.messagePayload.body.content,
-        images);
+        images
+    );
 
-    if (action.messagePayload.attachments && action.messagePayload.attachments.length > 0) {
+    if (
+        action.messagePayload.attachments &&
+        action.messagePayload.attachments.length > 0
+    ) {
```
Comment (mdrichardson) [inline samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js:153]: Should be able to simplify to:

```suggestion
    if (action.messagePayload.attachments?.length > 0) {
```

File: samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js [hunk 3]
```diff
@@ -1,45 +1,134 @@
...
+                context,
+                context.activity.from.id
+            );
+            return member.name;
+        } catch (e) {
+            if (e.code === 'MemberNotFoundInConversation') {
+                context.sendActivity(MessageFactory.text('Member not found.'));
+                return e.code;
+            } 
+        throw e;
```
Comment (mdrichardson) [inline samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js:62]:
```suggestion
            if (e.code === 'MemberNotFoundInConversation') {
                context.sendActivity(MessageFactory.text('Member not found.'));
                return e.code;
            } 
            throw e;        
```

File: samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js [hunk 4]
```diff
@@ -1,45 +1,134 @@
...
+                return shareMessageCommand(context, action);
+            default:
+                throw new Error('NotImplemented');
         }
     }
+
+    async handleTeamsMessagingExtensionFetchTask(context, action) {
+        try {
+           const member = await this.getSingleMember(context);
+            return {
```
Comment (mdrichardson) [inline samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js:21]:
```suggestion
           const member = await this.getSingleMember(context);
           return {
```

**Review by mdrichardson — CHANGES_REQUESTED** [review]

**Review by mdrichardson — APPROVED** [review]

**Review by EricDahlvang — APPROVED** [review]
Thanks @Trinetra-MSFT !

**Pull request merged**